var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -576,11 +644,25 @@ def edit(self, spec, prefix):
                     libs += ["-lcufft", "-lcublas"]
 
             cuda_arch = spec.variants["cuda_arch"].value[0]
-            if cuda_arch:
-                gpuver = {"35": "K40", "37": "K80", "60": "P100", "70": "V100"}[cuda_arch]
+            gpuver = gpu_map[cuda_arch]
+            if cuda_arch == "35" and spec.satisfies("+cuda_arch_35_k20x"):
+                gpuver = "K20X"
 
-                if cuda_arch == "35" and spec.satisfies("+cuda_arch_35_k20x"):
-                    gpuver = "K20X"
+        if spec.satisfies(+rocm) and spec.satisfies("@2022:"):
```

Comment:
FIY, I think this is equivalent to something like

```suggestion
        if "@2022: +rocm" in spec:
```